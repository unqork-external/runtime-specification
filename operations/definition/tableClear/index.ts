export * from './tableClear.model'
export * from './tableClear.options'
