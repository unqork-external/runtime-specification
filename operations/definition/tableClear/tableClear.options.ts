import { Description, Example, Required } from '@tsed/schema'

export class TableClearOperationOptions {
  @Required()
  @Description('Key of the Table component to apply the clear operation')
  public targetKey: string
}
