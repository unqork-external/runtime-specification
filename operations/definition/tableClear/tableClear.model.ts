import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { TableClearOperationOptions } from './tableClear.options'
import { OpsBuilderStable, Stable } from '../../../../decorators/stability'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.TABLE_CLEAR)
@Description('Clears data for the Table component.')
@Stable()
@OpsBuilderStable()
export class TableClearOperation extends Operation<OperationTypes.TABLE_CLEAR> {
  public type = OperationTypes.TABLE_CLEAR as const

  @Required()
  public options: TableClearOperationOptions = new TableClearOperationOptions()
}
