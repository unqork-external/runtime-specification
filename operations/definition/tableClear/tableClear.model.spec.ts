import { TableClearOperation } from './tableClear.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: TableClear Tests', () => {
  let schema

  beforeAll(() => {
    const response = generateSchemaAndValidate(TableClearOperation)
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
