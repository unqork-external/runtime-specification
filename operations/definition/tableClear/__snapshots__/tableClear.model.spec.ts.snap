// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: TableClear Tests should match this JSON Schema 1`] = `
{
  "definitions": {
    "TableClearOperationOptions": {
      "properties": {
        "targetKey": {
          "description": "Key of the Table component to apply the clear operation",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
      ],
      "type": "object",
    },
  },
  "description": "Clears data for the Table component.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/TableClearOperationOptions",
    },
    "type": {
      "const": "TABLE_CLEAR",
      "examples": [
        "TABLE_CLEAR",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "stability": "STABLE",
  "type": "object",
}
`;
